server/src/lib.rs

```diff
@@ -243,19 +288,30 @@ fn log_response(addr: &SocketAddr, method: &str, uri: &str, start: &Instant, sta
     );
 }
 
+
 async fn handle(
     server: ServerSettings,
     collector: Collector,
     db: Db,
     subscriptions: Subscriptions,
     heartbeat_tx: mpsc::Sender<WriteHeartbeatMessage>,
-    auth_mech: AuthenticationMechanism,
-    conn_state: Arc<Mutex<kerberos::State>>,
+    auth_infos: Option<AuthenticationInfos>,
```

Comment:
Not sure about this being an Option. I would rather see the "content" of the Tls variant to be optional or something.